csrc/polymorphic_value.h

```diff
@@ -294,13 +294,14 @@ inline PolymorphicValue erf(const PolymorphicValue& a) {
 }
 
 // Convert scalars, vector of scalars, vector of vector of scalars, etc., into
-// an at::Tensor
-inline PolymorphicValue toTensor(const PolymorphicValue& x) {
+// an at::Tensor. device argument allows for the creation of CPU Scalars.
+inline PolymorphicValue toTensor(
+    const PolymorphicValue& x,
+    c10::DeviceType device = at::kCUDA) {
   if (x.is<at::Tensor>()) {
     return x;
   }
-  // TODO: allow specifying device
-  auto options = at::TensorOptions().device(at::kCUDA, 0);
+  auto options = at::TensorOptions().device(device, 0);
```

Comment:
You are right that being always on device 0 "doesn't feel quite right". The only reason I hardcoded device 0 here is because this was only used to evaluate the expected value for `testValidate`, which always use device 0 today. I would recommend to change this method to accept `at::Device` instead of `at::DeviceType` if we are supporting multiple device types.

Reply:
I added an `int8_t` argument for the device index.